operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/polling/PerResourcePollingEventSource.java

```diff
@@ -43,19 +44,49 @@
   private final long period;
   private final Set<ResourceID> fetchedForPrimaries = ConcurrentHashMap.newKeySet();
 
+  public PerResourcePollingEventSource(ResourceFetcher<R, P> resourceFetcher,
+      EventSourceContext<P> context, Duration defaultPollingPeriod,
+      Class<R> resourceClass) {
+    this(resourceFetcher, context.getPrimaryCache(), defaultPollingPeriod.toMillis(),
+        null, resourceClass,
+        CacheKeyMapper.singleResourceCacheKeyMapper());
+  }
 
+  @Deprecated(forRemoval = true)
```

Comment:
Specify in the javadoc using the @deprecated attribute what is supposed to be used instead.